kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,47 @@
+import os
+import xmlschema
+import logging
+
+from .report import Report
+from .record import PROBLEM, Record, INFORMATION
+
+xml_schema = "./xml_schema/"
+LOGGER = logging.getLogger(__name__)
+
+
+def schemas(report: Report, parsed_xml):
+    metadatacount = 0
+    validcount = 1
+    for extension in parsed_xml.findall("extension"):
+        extension_point = (extension.get("point")).split(".")[-1]
+        if extension_point == "metadata":
+            metadatacount += 1
+
+        schema_path = xml_schema + extension_point + '.xsd'
+        if not os.path.exists(schema_path):
+            report.add(Record(PROBLEM, "%s doesn't exist" % extension_point))
+            continue
+
+        if not validate(report, extension, schema_path):
+            report.add(Record(PROBLEM, "Schema validation failed for %s" % extension_point))
+            validcount -= 1
+        parsed_xml.remove(extension)
+
+    root_schema = xml_schema + 'addon.xsd'
+    validate(report, parsed_xml, root_schema)
+
+    if metadatacount != 1:
+        report.add(Record(PROBLEM, 'metadata missing/occurred more then once'))
+
+    if validcount == 1:
+        report.add(Record(INFORMATION, "Valid XML file found"))
+
+
+def validate(report: Report, xml, schemapath):
+    try:
+        schema = xmlschema.XMLSchema(schemapath)
+        schema.validate(xml)
+        return True
+    except xmlschema.XMLSchemaValidationError as schemaerror:
+        print(schemaerror)
```

Comment:
You're passing a report as the argument, why do you do a print? What if we want to use another reporter in the future?

Reply:
I forgot to remove report from the argument. I have used print because if we add this in our report as a `PROBLEM` that will increase the count of errors (`ERROR: We found 1 problems and 0 warnings, please check the logfile.`) twice for a same problem.

Reply:
Agreed. Please remove the reporter from the method arguments if not used